src/jaegertracing/Tracer.cpp

```diff
@@ -101,7 +110,7 @@ Tracer::StartSpanWithOptions(string_view operationName,
         }
         else {
             const auto traceID = parent->traceID();
-            const auto spanID = randomID();
+            const auto spanID = self ? self->spanID() : randomID();
```

Comment:
this looks inconsistent. How do you rationalize only overriding span id, but not the trace id?

I think I would propose that SelfRef is only used as a lone reference for root span.

Reply:
> I think I would propose that SelfRef is only used as a lone reference for root span.
What if the library user wants to generate all spanIDs by themselves? I think it would be inconsistent, to let users generate spanIDs for root spans, but not for children.

Reply:
SelfRef can only be used for root spans, using them with any other reference will make StartSpanWithOptions return nullptr